boards/arm/efr32xg24_dk2601b/Kconfig.defconfig

```diff
@@ -25,6 +25,13 @@ config FPU
 
 endif # SOC_GECKO_USE_RAIL
 
+if SOC_GECKO_USE_RAIL_API
+
+config SOC_GECKO_USE_RAIL
+	default y
+
+endif # SOC_GECKO_USE_RAIL_API
```

Comment:
Hi @zohavas,
What is the purpose of having `SOC_GECKO_USE_RAIL` and `SOC_GECKO_USE_RAIL_API`?

Reply:
I will update the naming to PROPRIETARY as it will be more user friendly. 
These files will be only needed if user want to use the radio for custom configured phys.
BLE won't need these files so this is why there is an extra layer.

Reply:
Ok, I see.

However, I don't know if `PROPRIETARY` is meaningful for most of users. Maybe SOC_GECKO_CUSTOM_RADIO  or SOC_GECKO_USE_RAIL_CUSTOM would be make more sense? (BTW, maybe `SOC_GECKO_USE_RAIL` should be named `SOC_GECKO_RAIL`). @fkokosinski, I think you have a better knowledge of the Zephyr habits. Do you have an opinion?

Reply:
I don't really have a strong preference here, but I agree that `PRIOPIETARY` doesn't really say much. 

As for `SOC_GECKO_CUSTOM_RADIO`, I think we could even go a bit further and be more descriptive with `SOC_GECKO_CUSTOM_RADIO_PHY`. It's not like it will cost us a lot to type 4 more characters once and then copy it, and I suppose it will allow us to avoid confusion later on.